File: calc_pi.py
if __name__ == "__main__":
    N = 1E7
    x, y, hits = 0, 0, 0
    while x * x < N:
        y = 0
        while y * y < N:
            if x * x + y * y < N:
                hits += 1
            y += 1
        x += 1
    print(f"{(hits / N) * 4}")

File: calc_pi.c
#include <stdio.h>

int main(void)
{
    const int N = 1E7;
    double x, y, hits = 0;
    for (x = 0; x * x < N; ++x)
    {
        for (y = 0; y * y < N; ++y)
        {
            if (x * x + y * y < N)
            {
                ++hits;
            }
        }
    }
    printf("%lf\n", (hits / N) * 4);

    return 0;
}
